tracing/tracing/src/main/java/io/helidon/tracing/SpanListener.java

```diff
@@ -0,0 +1,144 @@
+ *         activated} †</td>
+ *         <td>After a span has been activated, creating a new scope in the process. </td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #closed(io.helidon.tracing.Span, io.helidon.tracing.Scope) closed} †</td>
+ *         <td>After a scope has been closed.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #ended(io.helidon.tracing.Span) ended (successful)} *</td>
+ *         <td>After a span has ended successfully.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #ended(io.helidon.tracing.Span, Throwable) ended (unsuccessful)} *</td>
+ *         <td>After a span has ended unsuccessfully. </td>
+ *     </tr>
+ *     </tbody>
+ *      <tfoot>
+ *         <tr>
+ *             <td colspan="2">† Not all spans are activated; it is up to the application or library code that creates and manages the span.
+ *              As a result Helidon might not invoke your listener's {@code activated} and {@code closed} methods for
+ *              every span.
+ *              <p>
+ *              * The successful or unsuccessful nature of a span's end is not about whether the tracing or telemetry
+ *              system failed to end the span. Rather, it indicates whether the code that ended the span indicated some error in
+ *              the processing which the span represents.
+ *             </td>
+ *         </tr>
+ *     </tfoot>
+ * </table>
+ * <p>
+ *     When Helidon invokes the listener methods it passes implementations of the key span types ({@link Span.Builder},
+ *     {@link Span}, {@link io.helidon.tracing.Scope}) which <em>do not</em> support lifecycle state changes.
+ *     If a listener tries to start or end or activate a span,
+ *     for example, Helidon throws an {@link java.lang.UnsupportedOperationException}.
+ */
+public interface SpanListener {
+
+    /**
+     * Invoked just prior to a {@linkplain io.helidon.tracing.Span span} being started from its
+     * {@linkplain io.helidon.tracing.Span.Builder builder}.
+     *
+     * @param spanBuilder the {@link Span.Builder} for the builder about to be used to start a span
+     * @throws java.lang.UnsupportedOperationException if the listener tries to start the span
+     */
+    default void starting(Span.Builder<?> spanBuilder) {
+    }
+
+    /**
+     * Invoked just after a {@linkplain io.helidon.tracing.Span span} has been started.
+     *
+     * @param span {@link io.helidon.tracing.Span} for the newly-started span
+     * @throws java.lang.UnsupportedOperationException if the listener tries to end the span
+     */
+    default void started(Span span) {
+    }
+
+    /**
+     * Invoked just after a {@linkplain io.helidon.tracing.Span span} has been activated.
+     * <p>
+     *     Callers should normally catch the {@link io.helidon.tracing.UnsupportedActivationException}, retrieve the
+     *     {@link io.helidon.tracing.Scope} from it, and then close that {@code Scope}.
```

Comment:
Reading probably too quickly, but is this still true? or was this changed with the log messages at warning level stuff?

Reply:
Nope, you're right. Fixed.